packages/shopify-codemod/transforms/split-return-assignments.js

```diff
@@ -0,0 +1,52 @@
+import {matchLast, forEachAssignment} from './utils';
+
+export default function splitReturnAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.Function, {
+      body: {
+        body: matchLast(j.ReturnStatement, {
+          type: 'ReturnStatement',
+          argument: j.AssignmentExpression,
+        }),
+      },
+    })
+    .forEach(({node: {body: {body}}}) => {
+      const returnLine = body[body.length - 1];
+      forEachAssignment(body[body.length - 1].argument, (assignment) => {
+        body.splice(body.indexOf(returnLine) + 1, 0, j.expressionStatement(assignment));
+      });
+      delete body[body.indexOf(returnLine)];
+      body.push(j.returnStatement(returnLine.argument.left));
```

Comment:
I'd still really prefer sticking to `NodePath`s instead if dropping down into the nodes, seems like it should be doable here.